askup/tests/test_functional/test_general.py

```diff
@@ -1583,21 +1575,15 @@ class TestSubscriptionsMailing(TestCase):
 
     fixtures = ['groups', 'mockup_data']
 
-    def setUp(self):
-        """
-        Set up the test conditions.
-        """
-        settings.DEBUG = False
-
     @client_user('student01', 'student01')
     @patch('askup.utils.general.do_send_subscriptions_to_recipients')
     def test_subscriptions_sending_no_actual_subscriptions(self, mock_do_send_subscriptions_to_recipients):
         """
         Test subscriptions sending.
         """
-        send_subscription_emails()
         mock_do_send_subscriptions_to_recipients.return_value = None
         mock_do_send_subscriptions_to_recipients.assert_not_called()
```

Comment:
The assert will have to come after the method under test, to ensure we're actually testing the functionality stated.

Reply:
That's right. But I've tested it 10 minutes ago and it seems like mock asserts are processing before the test method run. I've placed `send_subscription_emails()` in the top and in the bottom of the method. Changed it's assertion condition to `called` and `not_called` - it's working in both positions. :)
That's why in beforehand I didn't notice the order of those lines.

Reply:
@KhaimovMR `.assert_called()` does not work if it comes after the `send_subscription_emails()` call, which is good -- that's expected. So the test really needs to read:

```
send_subscription_emails()
mock_do_send_subscriptions_to_recipients.assert_not_called()
```

(the extra `return_value` step is irrelevant and can be omitted, both in this test case and the one below, `test_subscriptions_sending_with_subscriptions`)